include/envoy/api/api.h

```diff
@@ -36,9 +35,14 @@ class Api {
    * @return a reference to the Filesystem::Instance
    */
   virtual Filesystem::Instance& fileSystem() PURE;
+
+  /**
+   * @return a reference to the TimeSystem
+   */
+  virtual Event::TimeSystem& timeSystem() PURE;
```

Comment:
Per discussion, WDYT about just making this return a `TimeSource` in this PR? We can still take a `TimeSystem` in the constructor and not change anything else until a follow up, but this would get us closer to where we want to be?

Reply:
I think that would expand this PR significantly because there are a few places in the PR where an existing timeSystem() interface is implemented using API::timeSystem(). If that disappeared I'd have to change a bunch of call-sites. This would be trivial but it would make the PR a lot larger, and there is some semantic content here. So I'd prefer to do that in a separate PR. That OK?

I have this practice of trying to make semantically interesting PRs as small as possible, and letting trivial renaming or type-change PRs go large if needed.

Reply:
Sure SGTM. I think we are on the same page and heading in the right direction so :+1: